api/grpcserver/grpcserver_test.go

```diff
@@ -317,30 +315,37 @@ func NewTx(nonce uint64, recipient types.Address, signer *signing.EdSigner) *typ
 	return &tx
 }
 
-func newChallenge(nodeID types.NodeID, sequence uint64, prevAtxID, posAtxID types.ATXID, pubLayerID types.LayerID) types.NIPostChallenge {
-	challenge := types.NIPostChallenge{
-		NodeID:         nodeID,
+func newChallenge(sequence uint64, prevAtxID, posAtxID types.ATXID, pubLayerID types.LayerID) types.NIPostChallenge {
+	return types.NIPostChallenge{
 		Sequence:       sequence,
 		PrevATXID:      prevAtxID,
 		PubLayerID:     pubLayerID,
 		PositioningATX: posAtxID,
 	}
-	return challenge
-}
-
-func newAtx(challenge types.NIPostChallenge, nipost *types.NIPost, coinbase types.Address) *types.ActivationTx {
-	activationTx := &types.ActivationTx{
-		InnerActivationTx: types.InnerActivationTx{
-			ActivationTxHeader: types.ActivationTxHeader{
-				NIPostChallenge: challenge,
-				Coinbase:        coinbase,
-				NumUnits:        numUnits,
-			},
-			NIPost: nipost,
-		},
-	}
-	activationTx.CalcAndSetID()
-	return activationTx
+}
+
+func newAtx(challenge types.NIPostChallenge, sig *MockSigning, nipost *types.NIPost, numUnits uint, coinbase types.Address) *types.ActivationTx {
+	atx := types.NewActivationTx(challenge, coinbase, nipost, numUnits, nil)
+	activation.SignAtx(sig, atx)
+	atx.CalcAndSetID()
+	atx.CalcAndSetNodeID()
+	return atx
+}
+
+func NewMockSigner() *MockSigning {
```

Comment:
there is an implementation generated using gomock in signing module that can be used for this purpose

Reply:
Since this looks like a bigger effort and this PR is already quite large I opted to just add TODOs to refactor this in a future PR.